core/node.go

```diff
@@ -143,9 +198,51 @@ func (n *Node) ID() peer.ID {
 // Start causes the Node to continuously send messages to and receive messages
 // from its peers. It blocks until an error is encountered or `Stop` is called.
 func (n *Node) Start() error {
+	// Note: dht.Bootstrap has a somewhat confusing name. It doesn't automatically
+	// connect to the bootstrap peers. It just starts the background process of
+	// searching for new peers.
+	if err := n.dht.Bootstrap(n.ctx); err != nil {
+		return err
+	}
+
+	// Advertise ourselves for the purposes of peer discovery.
+	discovery.Advertise(n.ctx, n.routingDiscovery, n.config.RendezvousString, discovery.TTL(advertiseTTL))
+
+	if n.config.UseBootstrapList {
```

Comment:
You should probably move this code above the advertisement. If you're connected to nobody, we won't manage to advertise anywhere.

Reply:
@raulk The doc comment for `discovery.Advertise` states:

> Advertise is a utility function that persistently advertises a service through an Advertiser

"persistently" seems to imply that it will continuously advertise to connected peers. Is that not the case? Maybe we could update the docs to clarify?

Reply:
You’re right. I thought you were using Advertiser.Advertise.